### PR TITLE
Bug 1683890 - Runtime error when encoding Unicode string

### PR DESCRIPTION
**Pull request by jhadvig**
Wasn't really sure if we also update the *secrets.scenario.ts* here by testing some unicode characters, since it's a bit of an edge use-case.

/assign @spadgett

### CODE REVIEW

File: frontend/public/components/utils/index.tsx [hunk 1]
```diff
@@ -47,6 +47,7 @@ export * from './k8s-watcher';
 export * from './workload-pause';
 export * from './list-dropdown';
 export * from './status-icon';
+export * from './encode';
```
Comment (spadgett) [inline frontend/public/components/utils/index.tsx]: I'd call the package `base64`

File: frontend/public/components/utils/encode.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+// https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/btoa#Unicode_strings
+export const base64encode = str => {
+    return window.btoa(unescape(encodeURIComponent(str)));
```
Comment (spadgett) [inline frontend/public/components/utils/encode.tsx]: indentation

File: frontend/public/components/utils/index.tsx (hunk 1 above)
Comment (jhadvig) [reply frontend/public/components/utils/index.tsx]: I did originally, but thought that we might add different types of en/decoding and wanted to have them in single file... Although not sure what other will we add.

**Comment by jhadvig** [comment]
@spadgett I've update the PR, using the `js-base64` lib for encoding and decoding Unicode chars.
I know that we have been talking about properly encoding and decoding binary files, but figured out that it will take a bit more work. Other thing is that we are pretty close to release so would rather do it as follow on, if you dont mind.

**Comment by jhadvig** [comment]
```
Error: Failed expectation
    at Object.<anonymous> (/go/src/github.com/openshift/console/frontend/integration-tests/tests/operator-hub/operator-hub.scenario.ts:131:72)
    at step (/go/src/github.com/openshift/console/frontend/integration-tests/tests/operator-hub/operator-hub.scenario.ts:33:23)
    at Object.next (/go/src/github.com/openshift/console/frontend/integration-tests/tests/operator-hub/operator-hub.scenario.ts:14:53)
    at fulfilled (/go/src/github.com/openshift/console/frontend/integration-tests/tests/operator-hub/operator-hub.scenario.ts:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
   ```
/retest

**Comment by jhadvig** [comment]
```
NoSuchElementError: No element found using locator: by.cssContainingText(".co-resource-list__item", "etcd")
    at elementArrayFinder.getWebElements.then (/go/src/github.com/openshift/console/frontend/node_modules/protractor/built/element.js:814:27)
    at ManagedPromise.invokeCallback_ (/go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/promise.js:1376:14)
    at TaskQueue.execute_ (/go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/promise.js:3084:14)
    at TaskQueue.executeNext_ (/go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/promise.js:3067:27)
    at asyncRun (/go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/promise.js:2927:27)
    at /go/src/github.com/openshift/console/frontend/node_modules/selenium-webdriver/lib/promise.js:668:7
    at process._tickCallback (internal/process/next_tick.js:68:7)Error
    at ElementArrayFinder.applyAction_ (/go/src/github.com/openshift/console/frontend/node_modules/protractor/built/element.js:459:27)
    at ElementArrayFinder.(anonymous function).args [as isDisplayed] (/go/src/github.com/openshift/console/frontend/node_modules/protractor/built/element.js:91:29)
    at ElementFinder.(anonymous function).args [as isDisplayed] (/go/src/github.com/openshift/console/frontend/node_modules/protractor/built/element.js:831:22)
    at /go/src/github.com/openshift/console/frontend/integration-tests/tests/olm/catalog.scenario.ts:62:44
    at Set.forEach (<anonymous>)
    at Object.<anonymous> (/go/src/github.com/openshift/console/frontend/integration-tests/tests/olm/catalog.scenario.ts:61:23)
    at step (/go/src/github.com/openshift/console/frontend/integration-tests/tests/olm/catalog.scenario.ts:33:23)
    at Object.next (/go/src/github.com/openshift/console/frontend/integration-tests/tests/olm/catalog.scenario.ts:14:53)
    at fulfilled (/go/src/github.com/openshift/console/frontend/integration-tests/tests/olm/catalog.scenario.ts:5:58)
```
/retest

**Pull request merged**